### ISSUE TITLE
what's the meaning of function mask2d in rnn.utils?

### ISSUE THREAD

**Issue by pauking** [quark0/darts#108]
def mask2d(B, D, keep_prob, cuda=True):
    m = torch.floor(torch.rand(B, D) + keep_prob) / keep_prob ##1
    m = Variable(m, requires_grad=False)
    if cuda:
        m = m.cuda()
    return m

I cannt understand the meaning of function above, especially line1,can you explain some more?

**Comment by pauking** [comment]
it is dropout
